lib/fast-path.ts

```diff
@@ -305,238 +322,170 @@ FPp.needsParens = function(assumeExpressionContext) {
     return false;
   }
 
-  const name = this.getName();
-
   // If the value of this path is some child of a Node and not a Node
   // itself, then it doesn't need parentheses. Only Node objects (in fact,
   // only Expression nodes) need parentheses.
   if (this.getValue() !== node) {
     return false;
   }
 
-  // Only statements don't need parentheses.
-  if (n.Statement.check(node)) {
-    return false;
-  }
```

Comment:
This style of check is important, since `Statement` is an abstract supertype with many concrete subtypes. Just checking `node.type === "Statement"` is not the same. I'll put this back, but just wanted you to be aware of the difference.

Reply:
Aaaaaah OK right. Thanks!